internal/core/pat/pat_service.go

```diff
@@ -0,0 +1,151 @@
+package pat
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"math/big"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/yaronf/httpsign"
+)
+
+type PatService struct {
+	patService repository
+}
+
+func NewPatService(repository repository) *PatService {
+	return &PatService{
+		patService: repository,
+	}
+}
+
+func hexPrivKeyToPubKey(hexPrivKey string) (ecdsa.PublicKey, error) {
+	privKeyD := new(big.Int)
+	_, err := privKeyD.SetString(hexPrivKey, 16)
+	if !err {
+		return ecdsa.PublicKey{}, fmt.Errorf("could not parse hexPrivKey")
+	}
+
+	privKey := &ecdsa.PrivateKey{
+		PublicKey: ecdsa.PublicKey{
+			Curve: elliptic.P256(),
+			X:     new(big.Int),
+			Y:     new(big.Int),
+		},
+		D: privKeyD,
+	}
+	privKey.PublicKey.X, privKey.PublicKey.Y = privKey.PublicKey.Curve.ScalarBaseMult(privKey.D.Bytes())
+
+	pubKey := &privKey.PublicKey
+	return *pubKey, nil
+}
+
+func PubKeyToFingerprint(pubKey string) (string, error) {
```

Comment:
I think this method should not be exported - as I see, you are currently only using it inside the pat package.